src/stores/grid_renderer_store.ts

```diff
@@ -347,48 +362,18 @@ export class GridRenderer {
         ctx.clip();
 
         const iconSize = icon.size;
-        const iconY = this.computeTextYCoordinate(box, iconSize);
-
-        const svg = icon.svg;
-        let x: number;
-        if (icon.horizontalAlign === "left") {
-          x = box.x + icon.margin;
-        } else if (icon.horizontalAlign === "right") {
-          x = box.x + box.width - iconSize - icon.margin;
-        } else {
-          x = box.x + (box.width - iconSize) / 2;
-        }
-        ctx.translate(x, iconY);
+        const { x, y } = this.getters.getCellIconRect(icon);
+        ctx.translate(x, y);
         ctx.scale(iconSize / svg.width, iconSize / svg.height);
-        ctx.fillStyle = svg.fillColor;
-        ctx.fill(new Path2D(svg.path));
+        for (const path of svg.paths) {
+          ctx.fillStyle = path.fillColor;
+          ctx.fill(new Path2D(path.path));
```

Comment:
could be cached or moved to each svg path. It looks like it's the first CPU consumer :deciduous_tree: 
![image](https://github.com/user-attachments/assets/5ddf569f-d107-46f4-b92e-50969d625580)
(profile when I was scrolling on a sheet full of checkboxes)